### PR TITLE
RCHAIN-3698: Add test that shows registry lookup is non-conflicting

### PR DESCRIPTION
**Pull request by tymm**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by tymm** [comment]
This is blocked by: https://github.com/rchain/rchain/pull/2692

**Comment by lukasz-golebiewski** [comment]
Needs https://github.com/rchain/rchain/pull/2739

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala [hunk 1]
```diff
@@ -67,6 +73,32 @@ class MultiParentCasperMergeSpec
     merges(echoContract(1), echoContract(2), Rho("Nil"))
   }
 
+  it should "not conflict on registry lookups" in effectTest {
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala]: This is good to be merged with `pendingUntilFixed`

**Review by lukasz-golebiewski — APPROVED** [review]
Nice one!

**Review by dzajkowski — APPROVED** [review]

**Pull request merged**